src/bp/sdk/botpress.d.ts

```diff
@@ -457,6 +457,71 @@ declare module 'botpress/sdk' {
   }
 
   export namespace NLU {
+    export class Engine {
+      static initialize: (config: Config, logger: NLU.Logger) => Promise<void>
+      static getHealth: () => Health
+      static getLanguages: () => string[]
+      constructor(defaultLanguage: string, botId: string, logger: Logger)
+      computeModelHash(intents: NLU.IntentDefinition[], entities: NLU.EntityDefinition[], lang: string): string
+      loadModel: (m: Model) => Promise<void>
+      train: (
+        intentDefs: NLU.IntentDefinition[],
+        entityDefs: NLU.EntityDefinition[],
+        languageCode: string,
+        reportTrainingProgress?: ProgressReporter,
```

Comment:
should this ProgressReporter be part of the tools ?

Reply:
Nah, the tools have to be built by the engine inside nlu-core. This report cb has to be passed by the wrapper.